tests/client/net_handler_client.c

```diff
@@ -622,6 +565,19 @@ void send_device_subs(dev_subs_t* subs, int num_devices) {
     usleep(400000);  // allow time for net handler and runtime to react and generate output before returning to client
 }
 
+DevData* get_next_dev_data() {
+    pthread_mutex_lock(&device_data_lock);
+    // We need to copy the data in the device_data Protobuf to return to the caller
+    // The easiest way I found was to pack it into a buffer then unpack it again, but this may not be optimal
```

Comment:
don't know if this an old comment but if we just want a copy of the device data, you can do use the function `memcpy` to copy bytes from one piece of memory to another; see https://www.tutorialspoint.com/c_standard_library/c_function_memcpy.htm